### PR TITLE
fix(front-matter): remove os specific newline

### PR DESCRIPTION
**Pull request by timreichen**
**Changes**
This PR removes the os specific check for `\r?` before `\n`.

**Reasoning**
This is the only occurrence where a os check is done. Even the `recognize()` function in the same module doesn't do it:
https://github.com/denoland/std/blob/99200dcd9039eea515f3f850b2f13cf7be585d03/front_matter/_shared.ts#L36

### CODE REVIEW

**Review by iuioiua — APPROVED** [review]
The reasoning makes sense to me. But this is definitely not a chore. It's either a breaking change or refactor.

File: front_matter/_formats.ts [hunk 1]
```diff
@@ -21,9 +21,7 @@ const JSON_HEADER = `(---json|${JSON_DELIMITER})\\s*`;
 const JSON_FOOTER = `(?:---|${JSON_DELIMITER})`;
 
 const DATA = "([\\s\\S]+?)";
-const NEWLINE = `${
-  globalThis?.Deno?.build?.os === "windows" ? "\\r?" : ""
-}(?:\\n)?`;
+const NEWLINE = "\\r?\\n?";
```
Comment (iuioiua) [inline front_matter/_formats.ts:24]:
```suggestion
const NEWLINE = "\\r?(?:\\n)?";
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

File: front_matter/_formats.ts (hunk 1 above)
Comment (timreichen) [reply front_matter/_formats.ts:24]: I think these brackets don't do anything.

**Comment by kt3k** [comment]
I don't think this breaks any existing practical usage as the condition is now more permissive (windows-style new line is now allowed in linux and mac).

How about tagging this as 'fix'?

**Comment by iuioiua** [comment]
It's more permissive but changes expected behavior, no?

**Comment by kt3k** [comment]
If we change the parsing behavior depending on the platform, that means the users can't share the exact same markdown files between different platforms (e.g. If windows user creates a markdown file with front-matter with CRLF, then that file can't be parsed in mac/linux). I don't think that meets the expectation of the users.

File: front_matter/_formats.ts (hunk 1 above)
Comment (iuioiua) [reply front_matter/_formats.ts:24]: Let's merge as-is but keep this suggestion in mind in case anyone runs into any issues.

**Review by iuioiua — APPROVED** [review]

**Pull request merged**